src/chartkit/components/ChartWrapper.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import Component from 'react-component-component';
+
+import LoadingSpinner from 'uikit/LoadingSpinner';
+
+/**
+ * Expects an axios object to make a request
+ * an api endpoint
+ * optional data transform
+ * optional loading element
+ * optional error element
+ */
+const ChartWrapper = ({
+  api = '',
+  endpoint = '',
+  transform = x => x,
+  children,
+  Loader = LoadingSpinner,
+}) => (
+  <Component
+    style={{ height: '100%' }}
+    initialState={{ data: null, isLoading: true }}
+    didMount={({ setState }) => {
+      api(`${endpoint}`)
+        .then(resp => resp.data)
+        .then(data => transform(data))
+        .then(data => setState({ data: data, isLoading: false }))
+        .catch(err => console.log('err', err));
+    }}
+  >
+    {({ state }) =>
+      state.isLoading ? (
+        <Loader size="50px" />
+      ) : state.data ? (
+        React.cloneElement(children, { data: state.data })
```

Comment:
That's a fair point. Only reason I didn't go with render props was cleaner code imho. For instance a large page with 8 charts would end up having the same conditionals of:
loading => show loader
no data => show error message
data => show chart

Can we bump in this PR? ;D

Reply:
well we can keep the loading and error state in this component, but just pass along `data` to the children, so the consuming code looks like:
```
<ChartWrapper endpoint={...}>
  {({data}) => <SomeChart data={data}/>}
<ChartWrapper>
```
I think that's a good compromise.

In fact, if the only thing `SomeChart` needs is `data`, it can be simplified to:
`<ChartWrapper endpoint={...}> {SomeChart} </ChartWrapper>` or even`<ChartWrapper endpoint={...} children={SomeChart} />` if you're feeling fancy :))))

Reply:
another idea if you want to keep the loading and error state centralized without having it internal to `ChartWrapper`:
```
<ChartWrapper>
  {fetchState => (
    <LoadGate fetchState={fetchState} chartComponent={SomeChart} />
  )}
</ChartWrapper>

Reply:
Sounds great!